piccolo/apps/migrations/auto/migration_manager.py

```diff
@@ -534,6 +534,38 @@ async def _run_alter_columns(self, backwards: bool = False):
 
                 ###############################################################
 
+                on_delete = params.get("on_delete")
+                on_update = params.get("on_update")
+                if on_delete is not None or on_update is not None:
+                    # use to find FK table `referenced_table_name`` and
+                    # `referenced_column_name` for alter fk constraints
+                    existing_table = await self.get_table_from_snapshot(
+                        table_class_name=table_class_name,
+                        app_name=self.app_name,
+                    )
+                    # first drop fk constraint on alter table
+                    await self._run_query(
+                        _Table.alter().drop_foreign_key_constraint(
+                            column=alter_column.column_name
+                        )
+                    )
+                    # then add fk constraint on alter table
+                    for item in existing_table._meta.foreign_key_columns:
```

Comment:
I haven't had a chance to properly think about this yet - but why do we loop through all of the foreign key columns on the table?